bitbots_vision/src/bitbots_vision/vision_modules/field_boundary.py

```diff
@@ -629,13 +657,59 @@ def _calculate_field_boundary(_image, _field_color_detector, _x_steps, _y_steps,
         return _field_boundary_points
 
 
+class CVReversedFieldBoundaryAlgorithm(FieldBoundaryAlgorithm):
+    @staticmethod
+    def _calculate_field_boundary(image, field_color_detector, x_steps, y_steps, roi_height, roi_width, roi_increase, green_threshold):
+        # type: () -> list
+        """
+        Finds the points of the field boundary visible in the image. Uses the reversed method iterating from bottom to
+        top until it finds enough non green points. Useful for when two fields are adjacent to each other.
+
+        :returns: list of field boundary points
+        """
+        # calculate the field_mask which contains 0 for non-green pixels and 255 for green pixels in the image
+        # index counting up from top to bottom and left to right
+        field_mask = field_color_detector.get_mask_image()
+
+        # Scale the image down
+        subsampled_mask = cv2.resize(field_mask,(x_steps,y_steps), interpolation=cv2.INTER_AREA)
+        # Define the blur kernel
+        kernel = (2 * roi_height + 1, 2 * roi_width + 1)
```

Comment:
Why is this so large?

Reply:
The kernel? This why should it be large? The roi_height is normally a value of 1. The rest of the calculation is added, to get an odd result every time, which is needed for the blur. Also this calculation output is 1, which means no bluring for an roi size of 0 and 1 on each side for roi 1 and so on.

Reply:
First, "one on each side for roi of 1" would mean that a "height" or "width" of 1 leads to an actual height or width of 3, which is larger than 1. But I think that this is the case everywhere. Second, in this case the width of the kernel considers the subsampled pixels which significantly enlarges the corresponding area in the original image.

Reply:
Of cause the area is quite large after the downscaling. But the kernel normally only considers the direct neighbors by setting the value to 1. This works quite well on our settings and is much faster than bluring the original image. Higher values are nessesary if we increase the resolution further and need to blur bigger areas. Of cause height is a bit misleading if we a talk about some radius like thing, but an other parameter would break the interface quite a bit and this one was the closest one. 
If we want to use the real height/width, we could add an integer division on the height/width value first. That would also output only odd values for the kernel size, but parameter values like 2 and 3 would be equal.

Reply:
I added the integer division thing.